### ISSUE TITLE
lack the option "move" that appears when you right-click the mouse over running applications on the taskbar

### ISSUE THREAD

**Issue by aleex5** [lxde/lxqt#1434]
for me that option is very important and I use it a lot in games, I always configure the games in window mode without borders to save space in laptops

for example I always play league of legends in linux, by means of "wine staging" and I configure the window without borders, there are times that the window does not stay where I want and in those cases I use that option

I recently tested for the first time "lxqt" and my first impression was too good, it has good performance, good looks, it uses openbox as window manager (one of the most stable and lightweight that exists) and also is written in qt, that to me seem is one of the best framework that exist in linux

but I realized that you do not have that option, I leave an image with a red arrow to indicate what I mean (the image is from another desktop)

![resize_method_5](https://user-images.githubusercontent.com/35360938/34862718-94344648-f74a-11e7-8e43-2490d0174e86.png)

**Comment by tsujan** [comment]
I never use such menus but was surprised to see that only the "move" menu-item was missing from that nice menu of LXQt. Yes, it should be added.

**Comment by aleex5** [comment]
Each person is a world, we are all different
in my case I use the menu a lot
It would be great if they added those two options
In fact, I remember that in "lxde" those options were not added either

**Comment by tsujan** [comment]
@aleex5 I, for one, agree that it's important. I used the words "... but was surprised to see..." ;)

**Comment by agaida** [comment]
erm - please help me a little bit - but this is a native WM thing afaik - so not so much we can do i think

**Comment by tsujan** [comment]
@agaida Qt is able to move and resize windows. I think what's requested here should be quite possible.

**Comment by tsujan** [comment]
@agaida Strange! I can't find any KWindowSystem method for moving windows, let alone for resizing them! That's while, KWindowSystem can minimize, restore, move to current desktop... windows. I don't understand why it can't move windows. Maybe because of wayland.

It's easy to move windows under X11 -- and we use X11 -- but we rely on KWindowSystem. So, although this problem is fixable theoretically, its fix may not be feasible for now :(

**Comment by tsujan** [comment]
OK, KDE devs have already solved this problem and their solution should be independent of kwin because its works with openbox too.

**Comment by agaida** [comment]
What i wanted to show was that things differ - in regard who is paint the Window - and ofcourse the decorations.

**Comment by tsujan** [comment]
I added a preliminary move item and it works with Openbox: https://www.dropbox.com/s/w4wpzl4llatl6l0/move.mkv

However, I still don't know why it doesn't work with Compiz+Emerald -- my favorite compositor -- because, theoretically, it should work with all x11 WMs.

**Comment by agaida** [comment]
as long we can do this as generic as possible go for it

**Comment by tsujan** [comment]
@agaida Yes, it done by KWindowSystem but isn't ready yet. In the video, you could see a jump but I think it can be fixed. A resize item is also needed... The Compiz issue may be related to the fact that its code is old, although quite functional.

**Comment by agaida** [comment]
that would be really cool

**Comment by tsujan** [comment]
I think it's done: https://www.dropbox.com/s/w4wpzl4llatl6l0/move.mkv

It works fine both with Openbox and with Kwin. It's most smooth with Kwin; with Openbox, rarely a jump may happen on fast mouse movement but I think the code is correct (the video is taken under Openbox though). I haven't tested in with a dual monitor yet.

BTW, all slot connections have the old format in `lxqttaskbutton.cpp`. I think it's better to fix that first and in a separate PR.

**Comment by aleex5** [comment]
Wow
what a great job, I am quite surprised at the progress they made in such a short time

**Comment by tsujan** [comment]
@aleex5 
Luckily, it felt somehow entertaining ;) All we needed was already provided by `KWindowSystem`. It might take a while before a PR can be made because the current code should be cleaned up first, but the main job is already done.

**Comment by aleex5** [comment]
@tsujan 
what good news
I expect a lot these options
thanks for the great job they are doing

**Comment by tsujan** [comment]
Didn't have time to clean up the current code. The PR is at https://github.com/lxde/lxqt-panel/pull/452

@agaida Tests would be very appreciated.

**Comment by agaida** [comment]
Works fine for me - at least for xfwm4 - back in a few minutes - have to test openbox too

**Comment by tsujan** [comment]
Thanks! I didn't have xfwm4 here.

**Comment by agaida** [comment]
Openbox works fine too - i guess that you can test kwin alone - not installed here :P - next is flux

**Comment by stefonarch** [comment]
Move works fine here too, the resize option is new too no?

**Comment by agaida** [comment]
@tsujan - ok, i would describe fluxbox as "kind of works" - i guess we must not have full compat. The window handling in fluxbox is to different and ancient.

**Comment by tsujan** [comment]
@agaida It doesn't work under Compiz at all. I think `KWindowSystem` isn't fully compatible with some WMs. KDE Plasma should have used a similar code but I saw that KDE+Openbox moves windows jerkily.

**Comment by tsujan** [comment]
Or it can be said that some old WMs aren't fully covered by `KWindowSystem`

**Comment by agaida** [comment]
right, imho GTM

**Comment by tsujan** [comment]
> Move works fine here too, the resize option is new too no?

@stefonarch They were added in one commit because they used the same `KWindowSystem` function but in two different ways.

**Comment by tsujan** [comment]
@agaida We should really clean up the panel code later. Almost all signal-slot connections have the old Qt4 format. It'll be a lot of work but I might do it if I find more free time.

**Comment by agaida** [comment]
Full Ack
